### PR TITLE
Replace custom circular buffer with Apache commons(CircularFifoQueue) 

### LINKED ISSUE #1284: Get rid of airsonic's custom circular buffer implementation
Airsonic has its [own implementation]( https://github.com/airsonic/airsonic/blob/master/airsonic-main/src/main/java/org/airsonic/player/util/BoundedList.java ) of a circular buffer.

It would be nice to replace it with [either]( https://stackoverflow.com/questions/5498865/size-limited-queue-that-holds-last-n-elements-in-java ) the one from Apache Commons or Guava.

### PR DESCRIPTION
**Pull request by kessiler**
Closes #1284 
----

Guidelines for Contributing
---------------------------

Airsonic development is a community project, and contributions are welcomed. Here are a few guidelines you should follow before submitting:

  1.  **License Acceptance** All contributions must be licensed as [GNU GPLv3](https://github.com/airsonic/airsonic/blob/develop/LICENSE.txt) to be accepted. Use [`git commit --signoff`](https://jk.gs/git-commit.html) to acknowledge this.
  2.  **No Breakage** New features or changes to existing ones must not degrade the user experience. This means do not introduce bugs, remove functionality, or make large changes to existing themes/UI without prior discussion in an Issue.
  3.  **Coding standards** Language best-practices should be followed, comment generously, and avoid "clever" algorithms. Refactoring existing messes is great, but watch out for breakage.
  4.  **Be bold!** Without contributions, this project will vanish. If you just want to help out, try [submiting a patch](https://github.com/airsonic/airsonic/issues?q=is%3Aissue+is%3Aopen+label%3Apatches-welcome) for an open Issue.
  5.  **Stay relevant** Issues or commentary that is off-topic or tangential to Airsonic development is subject to moderation. Questions should be focused on improving documentation to solve a problem. Visit [Reddit](https://www.reddit.com/r/airsonic) or [IRC](http://webchat.freenode.net?channels=%23airsonic) for community discussion.

### CODE REVIEW

File: airsonic-main/src/main/java/org/airsonic/player/domain/TransferStatus.java [hunk 1]
```diff
@@ -298,5 +298,9 @@ public SampleHistory(SampleHistory other) {
             super(HISTORY_LENGTH);
             addAll(other);
         }
+
+        public Sample getLast() {
+            return this.get(this.size());
```
Comment (kessiler) [inline airsonic-main/src/main/java/org/airsonic/player/domain/TransferStatus.java]: Had to add this method as the data structure has changed to a queue

File: airsonic-main/pom.xml [hunk 2]
```diff
@@ -164,6 +164,12 @@
             <artifactId>commons-collections</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
```
Comment (jvoisin) [inline airsonic-main/pom.xml]: I'm not sure that we want to have two versions of the same package. Can't you use `commons-collections` instead?

Comment (kessiler) [reply airsonic-main/pom.xml]: CircularFifoQueue is just available in commons-collections4.
I left both packages due to incompatibility with the CollectionsUtils class.
Do you want to keep a single package? I can do that. I would rather keep collections4 which is more powerful.

Comment (jvoisin) [reply airsonic-main/pom.xml]: Keep the latest commons-collection one if possible then :)

Comment (kessiler) [reply airsonic-main/pom.xml]: @jvoisin should be good to go ;)

**Pull request merged**